src/api/procedures/modifyCorporateActionAgent.ts

```
@@ -0,0 +1,101 @@
+import { Identity, PolymeshError, Procedure, SecurityToken } from '~/internal';
+import { AuthorizationType, ErrorCode, RoleType, TxTags } from '~/types';
+import { ProcedureAuthorization, SignerType } from '~/types/internal';
+import {
+  authorizationToAuthorizationData,
+  dateToMoment,
+  signerToString,
+  signerValueToSignatory,
+} from '~/utils/conversion';
+
+export interface ModifyCorporateActionAgentParams {
+  target: string | Identity;
+  requestExpiry?: Date;
+}
+
+/**
+ * @hidden
+ */
+export type Params = { ticker: string } & ModifyCorporateActionAgentParams;
+
+/**
+ * @hidden
+ */
+export async function prepareModifyCorporateActionAgent(
```

Should also throw an error if trying to set the owner as the CAA